app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,55 @@
+class PostsController < ApplicationController
+  before_action :find_user
+  before_action :find_post, only: [:show]
+
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments).paginate(page: params[:page], per_page: 2)
+  end
+
+  def show; end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.create(title: params[:title], text: params[:text], author_id: current_user.id, comments_counter: 0,
+                        likes_counter: 0)
+    puts "post id is #{@post.id}"
+    if @post.save
+      redirect_to user_post_path(current_user, @post), notice: 'Post created successfully.'
+    else
+      render :ne
```

Comment:
- [ ] When I try to save a post is not saved. Kindly fix that. Your code here could use some improvement. When you use `.create` it runs the `.save` as well, so the `@post.save` on line 21 is not very needed. What we should do instead is to `build` the post and then save it. We should also use the `post_params` method you created.

  ```
  def create
    @post = current_user.posts.build(post_params)
    @post.comments_counter = 0
    @post.likes_counter = 0
    if @post.save
      redirect_to user_post_path(current_user, @post), notice: 'Post created successfully.'
    else
      render :new
    end
  end
  ```

- [ ] Also there is an error on this line. I think you meant `:new`

  ![image](https://github.com/hydaline-code/myblogapp/assets/81468675/459eff00-d3fe-4703-89d6-8537389015c6)